backend/src/main/java/ca/bc/gov/app/service/client/ClientLegacyService.java

```diff
@@ -307,6 +308,55 @@ public Flux<ForestClientDto> searchContact(ContactSearchDto dto) {
                     client.clientNumber())
             );
   }
+  
+  @SuppressWarnings("unchecked")
+  public Flux<ClientListDto> search(int page, int size, String keyword) {
+    log.info(
+        "Searching clients by keyword {} with page {} and size {}", 
+        keyword, 
+        page, 
+        size
+    );
+
+    return legacyApi
+        .get()
+        .uri(builder -> 
+              builder
+                .path("/api/search")
+                .queryParam("page", page)
+                .queryParam("size", size)
+                .queryParam("value", keyword)
+                .build(Map.of())
+        )
+        .exchangeToFlux(response -> {
+          List<String> totalCountHeader = response.headers().header("X-Total-Count");
+          Long count = totalCountHeader.isEmpty() ? 0L : Long.valueOf(totalCountHeader.get(0));
+
+          return response
+              .bodyToFlux(Map.class)
```

Comment:
Why it is using a map? It was supposed to be a DTO that reflects the one created on the legacy app

Reply:
Because in my DTO client name could be an individual name or a company name. In your DTO, you are aligning to the Old Client modeling, using the client name as last name. I can do this, but renaming some fields in your DTO.